cli/pipeline/template.go

```diff
@@ -132,7 +132,7 @@ spec:
         - "-c"
         - "mkdir -p $INPUT_PATH $OUTPUT_PATH &&
           {{ range $index, $input := .Step.Inputs }}
-          paddle data get {{ $input.Step }}/{{ $input.Version }} $INPUT_PATH -b {{ $input.Branch | sanitizeName }} -p {{ $input.Path }} {{ $input.Bucket | bucketParam }} &&
+          paddle data get {{ $input.Step }}/{{ $input.Version }} $INPUT_PATH -b {{ $input.Branch | sanitizeName }} -p {{ $input.Path }} -d {{ .Step.Subdir }} {{ $input.Bucket | bucketParam }} &&
```

Comment:
General question, what does the syntax `$input.Branch | sanitizeName` do exactly?

Reply:
This is specific to the [`text/template` package](https://godoc.org/text/template), rather than any Go magic. 

[This section of the docs](https://godoc.org/text/template#hdr-Pipelines) suggest the `|` is the pipeline operator, and takes the value on the left-hand-side and passes it to the function on the right-hand-side, and then takes the value returned from the function as the output. So I think this is taking `$input` (which is bound to each step because of the `{{ range }}` loop on the line above, getting the `Branch` field from the step, and then passing that to the `sanitizeName` function. I expect the `sanitizeName` function is something we've defined ourselves in this file.

Reply:
Ah yes that makes much more sense. It did seem like it was piping the output of one operation to another but I could only find references to the bitwise OR operator initially!